Fix `DEL` character in generated random strings

lrb2 opened the pull request:
### Description

Expands the prohibited ASCII characters for strings generated by `generateRandom()` to include the `DEL` character (127) in addition to the other control characters (0-31) and space (32).

Fixes #17054

williamdes reviewed (approved):
Awesome !

williamdes commented on libraries/classes/Util.php:
```diff
@@ -3018,7 +3018,7 @@ public static function generateRandom(int $length, bool $asHex = false): string
             $byte = ord((string) $random_func(1)) & 0x7f;
             // We want only ASCII chars
-            if ($byte <= 32) {
+            if ($byte <= 32 || $byte == 127) {
```
```suggestion
            if ($byte <= 32 || $byte === 127) {
```

williamdes commented on libraries/classes/Util.php:
```diff
@@ -3018,7 +3018,7 @@ public static function generateRandom(int $length, bool $asHex = false): string
             // to get ASCII only range
             $byte = ord((string) $random_func(1)) & 0x7f;
             // We want only ASCII chars
```
```suggestion
            // We want only ASCII chars and no DEL character (127)
```

lrb2 committed:
Fix `DEL` character in generated random strings
Apply suggestions from code review:
Use identical operator and improve comment

Signed-off-by: Lukas Bixen <lrb2_@outlook.com>

williamdes reviewed (approved):
💯
